Sparkle/SUDiskImageUnarchiver.m

```diff
@@ -89,12 +88,26 @@ - (void)extractDMGWithPassword:(NSString *)__unused password
         NSInteger taskResult = -1;
 		@try
 		{
-            NTSynchronousTask *task = [[NTSynchronousTask alloc] init];
-
-            [task run:@"/usr/bin/hdiutil" directory:@"/" withArgs:arguments input:promptData];
-
-            taskResult = [task result];
-            output = [[task output] copy];
+            NSTask *task = [[NSTask alloc] init];
+            task.launchPath = @"/usr/bin/hdiutil";
+            task.currentDirectoryPath = @"/";
+            task.arguments = arguments;
+            
+            NSPipe *inputPipe = [NSPipe pipe];
+            NSPipe *outputPipe = [NSPipe pipe];
+            
+            task.standardInput = inputPipe;
+            task.standardOutput = outputPipe;
+            
+            [task launch];
+            
+            [inputPipe.fileHandleForWriting writeData:promptData];
+            [inputPipe.fileHandleForWriting closeFile];
+            
+            [task waitUntilExit];
+            
+            output = [outputPipe.fileHandleForReading readDataToEndOfFile];
```

Comment:
I've found that such pattern can deadlock. If the task tries to write to stdout, but stdout buffer is full, it will wait forever, because your task is not reading yet.